packages/library-sync/src/commonMain/kotlin/io/realm/kotlin/mongodb/sync/SyncConfiguration.kt

```diff
@@ -283,6 +313,28 @@ public interface SyncConfiguration : Configuration {
         public fun errorHandler(errorHandler: SyncSession.ErrorHandler): Builder =
             apply { this.errorHandler = errorHandler }
 
+        /**
+         * Sets the strategy that would handle the client reset by this synced Realm.
+         *
+         * Flexible Sync applications only accept [ManuallyRecoverUnsyncedChangesStrategy] whereas
+         * partition-based applications only accept [DiscardUnsyncedChangesStrategy].
+         *
+         * In case that no strategy is defined it would use the default one defined in [Builder.build].
```

Comment:
We should indicate clearly what are the defaults strategy without referring to `Builder.build` this is already a public API and the best place to talk about these points ...